### ISSUE TITLE
Native compilation fails when referencing a (non-virtual) method overload declared in a parent class.

### ISSUE THREAD

**Issue by ImmutableOctet** [blitz-research/monkey2#204]
Okay, so this is kind of an odd C++ compilation issue: When attempting to reference a method overload declared in a parent class (Which is supposed to call a custom virtual version), specifically from the derived class's reference, compilation fails. Doing this from a reference to the *parent* class yields no errors.

I came across this when trying to implement my own 'Stream' classes, specifically, when I overrode the virtual version of the 'Write' method. After doing this, I was unable to use the 'Stream' class's

> Stream.Write:Int(data:DataBuffer, offset:Int, count:Int)

method (Non-virtual shorthand overload) from the context of *my custom stream class*, however, if I used a polymorphic reference to my object via the 'Stream' type, it behaves as expected.

### Below is an example that consistently fails at the native compilation phase using V1.1.06:

```
Function Main:Void()
	' To stop the native compilation error, switch these two lines:
	Local obj:Z
	'Local obj:X
	
	obj = New Z()
	
	obj.A()
End

Class X
	Method A:Void()
		A(10)
	End
	
	Method A:Void(value:Int) Virtual
		Print("X :: Value: " + value)
	End
End

Class Y Extends X
	Method A:Void(value:Int) Override
		Print("Y :: Value: " + value)
	End
End

Class Z Extends Y
	Method A:Void(value:Int) Override
		Print("Z :: Value: " + value)
	End
End
```

As you can see, the "default overload" of 'A' defined in 'X' is invalid when referenced by 'Y' or Z, but not when treating them as 'X'. This is inconsistent with the compiler's expectations, as it rightfully doesn't see a problem with this.

**To fix this problem at the code generation level, the compiler needs to generate 'using' statements for other (Non-virtual) overloads found in parent classes.**

### Here's a C++ example of this feature in practice, and how it needs to be implemented by the translator:

```
#include <iostream>

class X
{
	public:
		void A()
		{
			A(10);
		}

		virtual void A(int value)
		{
			std::cout << "X :: Value: " << value << '\n';
		}
};

class Y : public X
{
	public:
		// Comment this line in order to see the same compilation error.
		using X::A;

		virtual void A(int value) override
		{
			std::cout << "Y :: Value: " << value << '\n';
		}
};

int main()
{
	// This is the same as the Monkey 2 example,
	// except we have a using declaration in 'Y',
	// so we can use either an 'X' or 'Y' pointer.
	//X* obj;
	Y* obj;

	obj = new Y();

	obj->A();

	delete obj;

	return 0;
}
```

**TL;DR: MX2 needs to generate 'using' statements in order to inherit non-virtual methods when there's also virtual overloads of that method.**

**Comment by blitz-research** [comment]
I've just pushed a fix for this to the develop branch.

There was actually some 'using' logic in there, but it wasn't properly
handling super classes more than 1 level 'up'.

Thanks for the nice clean sample code!
